uberenv.py

```diff
@@ -687,50 +720,53 @@ def setup_paths_and_dirs(self):
                 spack_configs_path = pabs(new_path)
                 if not os.path.isdir(spack_configs_path):
                     print("[ERROR: Given path in 'spack_configs_path' does not exist: {0}]".format(spack_configs_path))
                     sys.exit(-1)
 
         # Set spack_env_directory to absolute path and (if exists) check validity
-        self.spack_env_name = self.opts["spack_env_name"]
+        self.spack_env_name = self.args["spack_env_name"]
         self.spack_env_directory = pabs(os.path.join(self.dest_dir, self.spack_env_name))
         if os.path.exists(self.spack_env_directory) and not os.path.isdir(self.spack_env_directory):
             print("[ERROR: Invalid Spack Environments directory. File given: {0} ]".format(self.spack_env_directory))
             sys.exit(-1)
 
         # Setup path of Spack Environment file if not specified on command line
         # Check under spack_config_path -> detected platform -> spack.yaml/ .lock
-        if self.opts["spack_env_file"] is None:
+        if self.args["spack_env_file"] is None:
             # Check if platform is detected
             uberenv_plat = self.detect_platform()
-            if uberenv_plat is None:
-                print("[ERROR: Could not detect platform. Supply Spack Environment file in command line with --spack-env-file=/path/to/spack.yaml")
-                sys.exit(-1)
-
-            # Check if a path to an init file is located
-            self.spack_env_file = pabs(pjoin(spack_configs_path, uberenv_plat))
-            spack_env_yaml = pjoin(self.spack_env_file, "spack.yaml")
-            spack_env_lock = pjoin(self.spack_env_file, "spack.lock")
-            if os.path.exists(spack_env_yaml):
-                self.spack_env_file = spack_env_yaml
-            elif os.path.exists(spack_env_lock):
-                self.spack_env_file = spack_env_lock
+            if not uberenv_plat is None:
+                # Check if a path to an init file is located
+                self.spack_env_file = pabs(pjoin(spack_configs_path, uberenv_plat))
+                spack_env_yaml = pjoin(self.spack_env_file, "spack.yaml")
+                spack_env_lock = pjoin(self.spack_env_file, "spack.lock")
+                if os.path.exists(spack_env_yaml):
+                    self.spack_env_file = spack_env_yaml
+                elif os.path.exists(spack_env_lock):
+                    self.spack_env_file = spack_env_lock
+                else:
+                    print("[WARNING: Could not find Spack Environment file (e.g. spack.yaml) under: {0}]".format(self.spack_env_file))
+                    self.spack_env_file = None
             else:
-                print("[ERROR: Could not find Spack Environment file (e.g. spack.yaml) under: {0}]".format(self.spack_env_file))
-                sys.exit(-1)
+                print("[WARNING: Could not detect platform. Supply Spack Environment file using the command line argument: --spack-env-file=/path/to/spack.yaml")
```

Comment:
Minor suggestion: 
Instead of the double negative, if might be easier to read by switching the order to
```python
  if uberenv_plat in None:
    print("[Warning: could not detect....")
  else:
    # check if a platform....
```

(Please ignore this suggestion if the warning goes away).